sdk/search/Azure.Search.Documents/src/Azure.Search.Documents.csproj

```diff
@@ -22,7 +22,6 @@
     <Compile Include="$(AzureCoreSharedSources)AzureEventSource.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)AzureKeyCredentialPolicy.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared\Core" />
-    <Compile Include="$(AzureCoreSharedSources)ObjectNotDisposedException.cs" LinkBase="Shared" />
```

Comment:
Are we the only ones using this? Of course it was Search-specific before as shared source, but I wonder if we shouldn't make it public in Azure.Core. It *could* be useful for any SDK and better to move now than later.

Reply:
It looks like Search is the only one using it right now, so if it's ok, I'd like to move it there temporarily.  We can add it to https://github.com/Azure/azure-sdk-for-net/issues/28369 as something to make public, and if you want to own and drive taking it through arch board, I think that would be great.  Let me know if you'd prefer to handle it differently!

Reply:
My apologies. I didn't realize it wasn't public already. Figured it was for callers to catch. Thank you for pointing that out.